### PR TITLE
Add download/upload operators for GCS and Google Sheets

### PR DESCRIPTION
**Pull request by turbaszek**
---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)
Depends on #7853 

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/suite/hooks/sheets.py [hunk 1]
```diff
@@ -375,3 +375,60 @@ def batch_clear(self, spreadsheet_id: str, ranges: List) -> Dict:
...
+            .get(spreadsheetId=spreadsheet_id)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def create_spreadsheet(self, spreadsheet: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Creates a spreadsheet, returning the newly created spreadsheet.
+
+        :param spreadsheet: an instance of Spreadsheet
```
Comment (michalslowikowski00) [inline airflow/providers/google/suite/hooks/sheets.py:421]: Above methods docstrings have `:return:`.

File: airflow/providers/google/suite/hooks/sheets.py [hunk 2]
```diff
@@ -375,3 +375,60 @@ def batch_clear(self, spreadsheet_id: str, ranges: List) -> Dict:
         ).execute(num_retries=self.num_retries)
 
         return response
+
+    def get_sheet_titles(self, spreadsheet_id: str, sheet_filter: Optional[List[str]] = None):
+        """
+        Retrieves the sheet titles from a spreadsheet matching the given id and sheet filter.
+
+        :param spreadsheet_id: The spreadsheet id.
```
Comment (michalslowikowski00) [inline airflow/providers/google/suite/hooks/sheets.py:383]: Why there are no :type param: here and within others methods?

File: airflow/providers/google/suite/operators/sheets.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+import csv
+from tempfile import NamedTemporaryFile
+from typing import Any, Dict, List, Optional
+
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.providers.google.suite.hooks.sheets import GSheetsHook
```
Comment (michalslowikowski00) [inline airflow/providers/google/suite/operators/sheets.py:24]: Wdyt about more explicit name like `GoogleSheetsHook`.

**Review by michalslowikowski00 — APPROVED** [review]
LGTM. :)

File: airflow/providers/google/suite/operators/sheets.py (hunk 3 above)
Comment (turbaszek) [reply airflow/providers/google/suite/operators/sheets.py:24]: I can refactor it, the hook already exists

File: airflow/providers/google/suite/hooks/sheets.py (hunk 2 above)
Comment (turbaszek) [reply airflow/providers/google/suite/hooks/sheets.py:383]: We have type annotations, however, I think I should add them for docs sake - @mik-laj ?

**Review by potiuk — APPROVED** [review]

File: airflow/providers/google/suite/operators/sheets.py (hunk 3 above)
Comment (turbaszek) [reply airflow/providers/google/suite/operators/sheets.py:24]: We can do it in next PR

**Comment by michalslowikowski00** [comment]
🚀

**Comment by potiuk** [comment]
Go!

**Pull request merged**